container/kvm/export_test.go

```diff
@@ -13,6 +18,41 @@ var (
 	TestStartParams = &startParams
 )
 
+func MakeTestableCreateMachineParams(params *CreateMachineParams, pathfinder func(string) (string, error), runCmd runFunc) {
+	params.pathfinder = pathfinder
+	params.runCmd = runCmd
+	return
+}
+
 func NewEmptyKvmContainer() *kvmContainer {
 	return &kvmContainer{}
 }
+
+func NewTestContainer(name string, runCmd runFunc, pathfinder func(string) (string, error)) *kvmContainer {
+	return &kvmContainer{name: name, runCmd: runCmd, pathfinder: pathfinder}
+}
+
+// NewRunStub is a stub to fake shelling out to os.Exec or utils.RunCommand.
+func NewRunStub(output string, err error) *runStub {
+	return &runStub{output: output, err: err}
+}
+
+type runStub struct {
+	output string
+	err    error
+	calls  []string
+}
+
+func (s *runStub) Run(cmd string, args ...string) (string, error) {
+	call := []string{cmd}
+	call = append(call, args...)
+	s.calls = append(s.calls, strings.Join(call, " "))
+	if s.err != nil {
+		return s.err.Error(), s.err
```

Comment:
Exactly that. The run func was pre-existing, and seems based on the utils.RunCommand which returns combined output.